mobile/fastlane/Fastfile

```diff
@@ -0,0 +1,172 @@
+# This file defines the signing and packaging lanes for iOS
+# Building is done separately via make targets
+
+default_platform(:ios)
+
+# Build configuration
+APP_NAME_RELEASE = "Status.app"
+APP_NAME_PR = "StatusPR.app"
+DISPLAY_NAME_RELEASE = "Status"
+DISPLAY_NAME_PR = "Status PR"
+PROJECT_DIR = File.expand_path("../", __dir__)
+BUILD_DIR = File.join(PROJECT_DIR, "bin", "ios", "qt6")
+
+platform :ios do
+  before_all do
+    UI.message("Project directory: #{PROJECT_DIR}")
+    UI.message("Build directory: #{BUILD_DIR}")
+  end
+
+  after_all do
+    # Clean up CI keychain after signing
+    if is_ci
+      delete_keychain(name: keychain_name) if File.exist?(File.expand_path("~/Library/Keychains/#{keychain_name}-db"))
+    end
+  end
+
+  error do
+    # Clean up CI keychain on failure too
+    if is_ci
+      delete_keychain(name: keychain_name) rescue nil
+    end
+  end
+
+  # ============================================
+  # PR Builds - Sign and package for ad-hoc distribution
+  # ============================================
+  desc "Sign and package iOS app for PRs"
+  lane :pr do
+    setup_ci_keychain
+
+    run_match(type: "adhoc")
+
+    resign_and_package(
+      app_name: APP_NAME_PR,
+      display_name: DISPLAY_NAME_PR,
+      profile_type: "adhoc"
+    )
+  end
+
+  # ============================================
+  # Release Builds - Sign and package for App Store
+  # ============================================
+  desc "Sign and package iOS app for release"
+  lane :release do
+    setup_ci_keychain
+
+    run_match(type: "appstore")
+
+    resign_and_package(
+      app_name: APP_NAME_RELEASE,
+      display_name: DISPLAY_NAME_RELEASE,
+      profile_type: "appstore"
+    )
+
+    if ENV["UPLOAD_TO_TESTFLIGHT"] == "true"
+      upload_to_testflight(
+        ipa: File.join(BUILD_DIR, "Status.ipa"),
+        skip_waiting_for_build_processing: true,
+        changelog: ENV["CHANGELOG"] || "New build from CI"
+      )
+    end
+  end
+
+  # ============================================
+  # Helper Methods
+  # ============================================
+
+  private_lane :setup_ci_keychain do
+    if is_ci
+      create_keychain(
+        name: keychain_name,
+        password: keychain_password,
+        default_keychain: true,
+        unlock: true,
+        timeout: 3600,
+        lock_when_sleeps: false
+      )
+    end
+  end
+
+  private_lane :run_match do |options|
+    match_params = {
+      type: options[:type],
+      readonly: false,
+      # Auto-regenerate profiles when new devices are registered (for dev and adhoc)
+      force_for_new_devices: options[:type] == "development" || options[:type] == "adhoc"
```

Comment:
I think I'll keep this only for "adhoc" will add option for "development" when we introduce dev certs later